### ISSUE TITLE
no_prefixes "is_" can sometimes be false positive

### ISSUE THREAD

**Issue by jan-jezek** [abaplint/abaplint#3197]
So..., interesting problem:
```abap
CLASS foo DEFINITION.

  PUBLIC SECTION.
    TYPES: BEGIN OF structure,
         value TYPE string,
       END OF structure.

    METHODS bar
      IMPORTING
        is_structure  TYPE structure
      CHANGING
        is_structure2 TYPE structure
        is_error      TYPE abap_bool.

ENDCLASS.
```
![image](https://github.com/abaplint/abaplint/assets/17530797/92a8a1c7-8b95-4efb-9ee5-6ab1db59e5e9)

For "is_structure" I would expect error here, so no problem here.
For "is_structure2" I cannot decide, the code is obviously poorly written, as if it would be hungarian notation it should be "cs_structure2".
For "is_error" I think this is a false positive as "is" is a verb in this case...

**Comment by DGRHicron** [comment]
Hi, I have the same issue in my projects. If I am not wrong Abaplint knows about parameter type. It could check if it is abap_bool and if it is then don't raise issue about is_ prefixes.

**Comment by larshp** [comment]
@jan-jezek @DGRHicron try again, 2.106.9 published

**Comment by DGRHicron** [comment]
@larshp, it works now for me. Thanks!
For future visitors, @larshp added new setting to the no_prefixes rule which is `"allowIsPrefixBoolean": true`.
